### PR TITLE
Fix open issues for next release

### PR DESCRIPTION
**Pull request by davidbuzinski**
This changes fixes the following issues:
https://github.com/mathworks/jenkins-matlab-plugin/issues/384
https://github.com/mathworks/jenkins-matlab-plugin/issues/380
https://github.com/mathworks/jenkins-matlab-plugin/issues/379

### CODE REVIEW

**Review by sameagen-MW — APPROVED** [review]
Idk what's going on with the Mac build failure, but the changes look good to me!

:shipit:

File: src/main/java/com/mathworks/ci/tools/MatlabInstaller.java [hunk 1]
```diff
@@ -70,48 +70,71 @@ public void setProducts(String products) {
...
-        args.add(destination.getParent().getRemote() + getNodeSpecificMPMExecutor(node));
+        args.add(mpmPath.getRemote());
         args.add("install");
         appendReleaseToArguments(release, args, log);
         args.add("--destination=" + destination.getRemote());
         addMatlabProductsToArgs(args, products);
-        installerProc.pwd(destination).cmds(args).stdout(log);
+
+        ByteArrayOutputStream err = new ByteArrayOutputStream();
+        installerProc.pwd(destination).cmds(args).stdout(err);
```
[the remark this answers is not in the document]
Comment (davidbuzinski) [reply src/main/java/com/mathworks/ci/tools/MatlabInstaller.java:122]: Oh yeah, good catch. That's why we weren't seeing the output during the builds yesterday!

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java [hunk 2]
```diff
@@ -197,20 +197,24 @@ public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher
...
-            context.env("PATH+matlab_batch", batchExecutable.getRemote());
+            context.env("PATH+MATLAB_BATCH", batchExecutable.getRemote());
         }
 
         // Add "matlabroot" without bin as env variable which will be available across
         // the build.
         context.env("matlabroot", nodeSpecificMatlab);
         // Add matlab bin to path to invoke MATLAB directly on command line.
-        context.env("PATH+matlabroot", matlabExecutablePath.getParent().getRemote());
+        context.env("PATH+MATLAB_ROOT", matlabBinDir.getRemote());
```
Comment (mw-kapilg) [inline src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java:214]: I'm not sure, should it be `MATLAB_ROOT` at line 212 as well?

File: src/main/java/com/mathworks/ci/Utilities.java [hunk 3]
```diff
@@ -43,13 +43,17 @@ public static void addMatlabToEnvPathFromAxis(Computer cmp, TaskListener listene
...
                 cmp.getNode(), listener, env);
 
-        if (matlabRoot != null && matlabRoot.getParent().exists()) {
-            env.put("PATH+matlab_batch", matlabRoot.getParent().getRemote());
+        FilePath toolHome = matlabRoot.getParent();
+        if (toolHome == null) {
+                return;
+        }
+        if (matlabRoot != null && toolHome.exists()) {
+            env.put("PATH+MATLAB_BATCH", toolHome.getRemote());
```
Comment (mw-kapilg) [inline src/main/java/com/mathworks/ci/Utilities.java:51]: We should have a `null` check for `matlabRoot` before calling `matlabRoot.getParent()`, right?

**Review by mw-kapilg — APPROVED** [review]
Just a couple of doubts, rest looks good to me!

File: src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java (hunk 2 above)
Comment (davidbuzinski) [reply src/main/java/com/mathworks/ci/UseMatlabVersionBuildWrapper.java:214]: I'm not sure if it actually matters, but I ended up reverting this change anyways (back to matlabroot).

File: src/main/java/com/mathworks/ci/Utilities.java (hunk 3 above)
Comment (davidbuzinski) [reply src/main/java/com/mathworks/ci/Utilities.java:51]: Added a null check for `matlabRoot`

**Pull request merged**